src/pages/ContentScripts/features/fast-pr/githubService.ts

```diff
@@ -0,0 +1,282 @@
+  });
+  if (!response.ok) {
+    return {
+      success: false,
+      message: t('error_get_file_sha', { status: response.status }),
+    };
+  }
+  const data = await response.json();
+  return {
+    success: true,
+    fileSha: data.sha || null,
+  };
+};
+//Create or update file content
+const createOrUpdateFileContent = async (
+  filePath: string,
+  content: string,
+  newBranch: string,
+  fileSha: string | null,
+  prRepo: string,
+  githubToken: string
+) => {
+  const response = await fetch(url.CREATE_OR_UPDATE_FILE_URL(filePath, prRepo), {
+    method: 'PUT',
+    headers: { Authorization: `Bearer ${githubToken}`, 'Content-Type': 'application/json' },
+    body: JSON.stringify({
+      message: COMMIT_MESSAGE(newBranch),
+      content: Buffer.from(content).toString('base64'),
+      branch: newBranch,
+      sha: fileSha,
+    }),
+  });
+  if (!response.ok) {
+    return {
+      success: false,
+      message: t('error_update_file_content', { status: response.status }),
+    };
+  }
+  return {
+    success: true,
+  };
+};
+//Create a new PR
+const createPullRequest = async (
+  prTitle: string,
+  prContent: string,
+  newBranch: string,
+  branch: string,
+  forkOwner: string | null,
+  originalRepo: string,
+  githubToken: string
+) => {
+  const head = forkOwner ? `${forkOwner}:${newBranch}` : newBranch;
+  const response = await fetch(url.CREATE_PULL_REQUEST_URL(originalRepo), {
+    method: 'POST',
+    headers: { Authorization: `Bearer ${githubToken}`, 'Content-Type': 'application/json' },
+    body: JSON.stringify({
+      title: prTitle,
+      body: prContent,
+      head: head,
+      base: branch,
+    }),
+  });
+  if (!response.ok) {
+    return {
+      success: false,
+      message: t('error_create_pr', { status: response.status }),
+    };
+  }
+  const data = await response.json();
+  return {
+    success: true,
+    html_url: data.html_url,
+  };
+};
+export const submitGithubPR = async (
+  form: FormInstance<any>,
+  originalRepo: string,
+  branch: string,
+  filePath: string,
+  fileContent: string
+) => {
+  const key = 'FastPR';
+  const values = await form.validateFields();
+  handleMessage('loading', t('status_get_github_token'), key);
+  const githubToken = await getGithubToken();
+  if (!githubToken) {
+    handleMessage('error', t('error_get_github_token'), key);
+    return;
+  }
+  const newBranch = generateBranchName();
+  handleMessage('loading', t('status_check_repo_permission'), key);
+  const permissionResult = await checkRepositoryPermission(originalRepo, githubToken);
+  if (!permissionResult.success && permissionResult.message) {
+    handleMessage('error', permissionResult.message, key);
+    return;
+  }
+  let prRepo = originalRepo;
+  let forkOwner: string | null = null;
+  if (!permissionResult.permission) {
+    handleMessage('loading', t('status_get_or_create_fork'), key);
+    const prRepoResult = await getOrCreateFork(originalRepo, githubToken);
```

Comment:
According to the documentation of the GitHub API, the create fork API is asynchronously, if we need to wait for several seconds here in case the fork process does not complete? Or if there is anyway to find out if the repo is a new forked one or an existing one, so we can just wait for a new fork one.

Reply:
At present, we are waiting for each step of the operation to be completed before proceeding to the next step.